Modules/_sre.c

```diff
@@ -1938,15 +1958,33 @@ _validate(PatternObject *self)
 /* -------------------------------------------------------------------- */
 /* match methods */
 
+static int
+match_traverse(MatchObject *self, visitproc visit, void *arg)
+{
+    Py_VISIT(self->pattern);
+    Py_VISIT(self->regs);
+    Py_VISIT(self->string);
+    Py_VISIT(Py_TYPE(self));
```

Comment:
> It must be called before the first DECREF(self), otherwise PyObject_GC_UnTrack() crash in dealloc.

Noted

> Usually, it's better to only track an object once it's fully initialized, but here it's more convient to do it here to be able to simply call DECREF() on error.

Yes, that was my thought when placing it.